File: src/TieFighter.java
import javax.swing.*;
import java.awt.*;
import java.util.Random;

/** Tie Fighter class for jWars project */
public class TieFighter extends Enemy {
	private ImageIcon imgIcon;

	/** Standard constructor takes in the lowest Y pos
	 *  @param lowestYpos lowest possible y pos */
	public TieFighter(int lowestYpos) {
		super(lowestYpos);
		this.setPointValue(5);
		imgIcon = new ImageIcon("tiefighter.gif");
		setEntityWidth(imgIcon.getIconWidth());
		setEntityHeight(imgIcon.getIconHeight());
	}

	/** Paint the tie fighter on the panel
	 *  @param g graphics object to paint on */
	public void paintComponent(Graphics g) {
		if(getXpos() == -1 && getYpos() == -1) {
			Random generator = new Random(System.nanoTime());
			setXpos(generator.nextInt(600) + 1);
			generator = new Random(System.nanoTime());
			setYpos(generator.nextInt(299) + 1);	
		}

		g.drawImage(imgIcon.getImage(), getXpos(), getYpos(), null, null);
	}
}

File: src/TieFighter.scala
import javax.swing._
import java.awt._
import scala.util.Random

/** Tie Fighter class for jWars project */
class TieFighter(lyp: Int) extends Enemy(lyp) {

	pointValue = 5
	private val imgIcon = new ImageIcon("tiefighter.gif")
	entityWidth = imgIcon.getIconWidth()
	entityHeight = imgIcon.getIconHeight()

	/** Paint the tie fighter on the panel
	 *  @param g graphics object to paint on */
	override def paintComponent(g: Graphics) = {
		if(xPos == -1 && yPos == -1) {
			xPos = Random.nextInt(600) + 1
			yPos = Random.nextInt(299) + 1
		}

		g.drawImage(imgIcon.getImage(), xPos, yPos, null, null)
		updateLocations()
	}
}
